docs/pages/server-access/guides/ec2-discovery.mdx

```diff
@@ -0,0 +1,146 @@
+
+## Step 2/4. Create an AWS Systems Manager Document
+
+Teleport makes use of AWS Systems Manager documents to execute commands necessary to
+install and start Teleport on discovered EC2 instances.
+
+A document with the following content should be created, named `TeleportDiscoveryInstaller`:
+
+```yaml
+---
+schemaVersion: '2.2'
+description: aws:runShellScript
+parameters:
+  token:
+    type: String
+    description: "(Required) The Teleport invite token to use when joining the cluster."
+  scriptName:
+    type: String
+    description: "(Required) The Teleport instalelr script to use when joining the cluster."
+mainSteps:
+- action: aws:downloadContent
+  name: downloadContent
+  inputs:
+    sourceType: "HTTP"
+    destinationPath: "/tmp/installTeleport.sh"
+    sourceInfo:
+      url: "https://teleport.example.com/webapi/scripts/{{ scriptName }}"
+- action: aws:runShellScript
+  name: runShellScript
+  inputs:
+    timeoutSeconds: '300'
+    runCommand:
+      - /bin/sh /tmp/installTeleport.sh "{{ token }}"
+```
+
+Here, `teleport.example.com` should be substituted with the public proxy URL of the
+Teleport cluster.
+
+When Teleport executes the Systems Manager document it will provide the `token`
+and `scriptName` parameters with defaults.
+
+- `{{ scriptName }}` will by default evaluate to `default-installer`
+- `{{ token }}` will by default evauluate to `aws-discovery-iam-token`
+
+
+## Step 3/4. Set up AWS IAM credentials
+
+In order for Teleport to send commands to EC2 instances, the node
+performing discovery requires an IAM policy with permissions to
+execute Systems Manager commands and retrieve EC2 instances:
+
+```js
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "ssm:SendCommand",
+                "ec2:DescribeInstances",
+                "ssm:GetCommandInvocation"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+All EC2 instances that are to be added to the Teleport cluster by the
+Discovery Node must include the `AmazonSSMManagedInstanceCore` IAM policy
+in order to receive commands from the Discovery Node. More information
+for IAM best practices on EC2 instances managed by Systems Manager can be found [here](https://aws.amazon.com/blogs/mt/applying-managed-instance-policy-best-practices/)
+
+## Step 4/4. Configure Teleport to discover EC2 instances
+
+In order to enable EC2 instance discovery the `discovery_service` must
+include an AWS section configured with at least one entry:
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  aws:
+   - types: ["ec2"]
+     regions: ["eu-central-1"]
+     tags:
+       "env": "prod" # Match EC2 instances where tag:env=prod
+```
+
+Full documentation on EC2 discovery configuration can be found [here](../../reference/config.mdx)
+
+Once Teleport is configured for discovery, it can be started and EC2
```

Comment:
I think we're missing a step that explains how to override the default install script. Maybe add an optional step "5" that explains what the default install script does and how to replace it if someone wants.